integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java

```diff
@@ -5,152 +5,185 @@
-                argumentParser.getNamespace(),
-                argumentParser.getJobName(),
-                argumentParser.getParentRunId()));
-        return argumentParser;
-      }
-    };
-  }
-
-  public static UUID getRandomUuid() {
-    return UUID.randomUUID();
+  @Builder.Default private String appName = null;
+  @Builder.Default private OpenLineageYaml openLineageYaml = new OpenLineageYaml();
+
+  public static ArgumentParser parse(SparkConf conf) {
+    ArgumentParserBuilder builder = ArgumentParser.builder();
+    adjustOldConfigs(conf);
+    conf.setIfMissing(SPARK_CONF_DISABLED_FACETS, DEFAULT_DISABLED_FACETS);
+    conf.setIfMissing(SPARK_CONF_TRANSPORT_TYPE, "http");
+
+    if (conf.get(SPARK_CONF_TRANSPORT_TYPE).equals("http")) {
+      findSparkConfigKey(conf, SPARK_CONF_HTTP_URL)
+          .ifPresent(url -> UrlParser.parseUrl(url).forEach(conf::set));
+    }
+    findSparkConfigKey(conf, SPARK_CONF_APP_NAME)
+        .filter(str -> !str.isEmpty())
+        .ifPresent(builder::appName);
+    findSparkConfigKey(conf, SPARK_CONF_NAMESPACE).ifPresent(builder::namespace);
+    findSparkConfigKey(conf, SPARK_CONF_JOB_NAME).ifPresent(builder::jobName);
+    findSparkConfigKey(conf, SPARK_CONF_PARENT_RUN_ID).ifPresent(builder::parentRunId);
+    builder.openLineageYaml(extractOpenlineageConfFromSparkConf(conf));
+    return builder.build();
   }
 
-  private static Optional<String> getNamedStringParameter(
-      List<NameValuePair> nameValuePairList, String name) {
-    return Optional.ofNullable(getNamedParameter(nameValuePairList, name))
-        .filter(StringUtils::isNoneBlank);
+  // adjust properties so the old pipelines are allowed
+  private static void adjustOldConfigs(SparkConf conf) {
+    findSparkConfigKey(conf, "spark.openlineage.host")
+        .ifPresent(
+            c -> {
+              replaceConfigEntry(conf, SPARK_CONF_HTTP_URL, c, "spark.openlineage.host");
+            });
+
+    findSparkConfigKey(conf, "spark.openlineage.timeout")
+        .ifPresent(
+            c -> {
+              replaceConfigEntry(
+                  conf, UrlParser.SPARK_CONF_TIMEOUT, c, "spark.openlineage.timeout");
+            });
+
+    findSparkConfigKey(conf, "spark.openlineage.version")
+        .ifPresent(
+            c -> {
+              replaceConfigEntry(
+                  conf,
+                  UrlParser.SPARK_CONF_API_ENDPOINT,
+                  String.format("api/v%s/lineage", c),
+                  "spark.openlineage.version");
+            });
+
+    findSparkConfigKey(conf, "spark.openlineage.apiKey")
+        .ifPresent(
+            c -> {
+              conf.setIfMissing(UrlParser.SPARK_CONF_API_KEY, c);
+              replaceConfigEntry(
+                  conf, UrlParser.SPARK_CONF_AUTH_TYPE, "api_key", "spark.openlineage.apiKey");
+            });
+
+    findSparkConfigKey(conf, "spark.openlineage.consoleTransport")
+        .ifPresent(
+            c -> {
+              if (c.equals("true")) {
```

Comment:
* consider using `filter` before `ifPresent`
 * `equalsIgnoreCase` may be a better option for string comparison.